### PR TITLE
feat: support pdb with v1 and v1beta version to control pod's deletion

### PR DESCRIPTION
**Pull request by jiuker**
support pdb

- [x] create tenant
- [x] expand tenant
- [x] decomm pool from tenant
- [x] delete tenant

Test steps:
1.start a tenant (4*4)
2. wait for tenant's status are ready
3.`kubectl get pdb -A` you will see that.
4. And delete the all pod. You will see delete the pod one by one.(For MinAvailable are 3)

### CODE REVIEW

File: pkg/controller/pdb.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+// thread safe
+func (c *Controller) GetPDBAvailable() PDBAvailable {
+	globalPDBAvailableOnce.Do(func() {
+		defer func() {
+			if globalPDBAvailable.v1 {
+				klog.Infof("PodDisruptionBudget: v1")
+			} else if globalPDBAvailable.v1beta {
+				klog.Infof("PodDisruptionBudget: v1beta")
+			} else {
+				klog.Infof("PodDisruptionBudget: unsupport")
```
Comment (shtripat) [inline pkg/controller/pdb.go:263]: `%s/unsupport/not supported/g` may be ?

File: pkg/controller/pdb.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+				return err
+			}
+			if existingStatefulSet.Status.ReadyReplicas != existingStatefulSet.Status.Replicas || existingStatefulSet.Status.Replicas == 0 {
+				return nil
+			}
+			if t.Status.CurrentState != StatusInitialized {
+				return nil
+			}
+		}
+		if available.V1Available() {
```
Comment (shtripat) [inline pkg/controller/pdb.go:99]: Logic is same for both cases (other than pdb object). Can we create a function and use for both scenarios?

Comment (jiuker) [reply pkg/controller/pdb.go:99]: I've tried it and can't implement it, the Client of these two is different, and the similarity of the incoming parameter is also different. @shtripat If you have a good thought you can share it with me as well

Comment (shtripat) [reply pkg/controller/pdb.go:99]: yes, after closure look, it doesn't look feasible. anyway its not a major one.

**Comment by shtripat** [comment]
@jiuker please add verification steps

**Comment by shtripat** [comment]
I followed the below steps
1. `TAG=minio/operator:pdb make all`
2. `kind create cluster --config kind-config.yaml`
3. `kind load docker-image docker.io/minio/operator:pdb`
4. `k kustomize . > operator-new.yaml`
5. Changed the operator image value in operator-new.yaml with `docker.io/minio/operator:pdb`
6. `k apply -f operator-new.yaml`
7. Once operator pods ready, created a tenant from operator console
8. `k logs minio-operator-7db874b9b-4pw9z -n minio-operator` shows error as below

```
I0718 11:00:37.918185       1 pdb.go:259] PodDisruptionBudget: v1
E0718 11:00:37.921171       1 main-controller.go:676] error syncing 'tenant-ns/tenant-1': poddisruptionbudgets.policy "tenant-1-pool-0" is forbidden: User "system:serviceaccount:minio-operator:minio-operator" cannot get resource "poddisruptionbudgets" in API group "policy" in the namespace "tenant-ns"
```

Also there are no `pdb` created

```
$ k get pdb -A
No resources found
```

**Comment by jiuker** [comment]
@shtripat I have fixed the `ClusterRole`. Test again plz. And you should update it.

File: helm/operator/templates/cluster-role.yaml [hunk 3]
```diff
@@ -164,3 +164,14 @@ rules:
       - get
       - update
       - create
+  - apiGroups:
```
Comment (shtripat) [inline helm/operator/templates/cluster-role.yaml:167]: These changes need to be added to `resources/base/cluster-role.yaml` as well.

Comment (jiuker) [reply helm/operator/templates/cluster-role.yaml:167]: added via https://github.com/minio/operator/pull/1681/commits/f049a4c0b47b03ba805ba244886a48bffa048721

**Comment by shtripat** [comment]
Below scenarios work fine
1. Create tenant
2. Delete tenant
3. Expand tenant

And PDBs get created for pools of the tenant.

Tried below scenario manually
1. Edited tenant and removed one pool from the list
2. deleted the stateful set for corresponding pool
3. PDB doesnt get deleted and error in operator pod ` error syncing 'tenant-ns/tenant-1': poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:minio-operator:minio-operator" cannot deletecollection resource "poddisruptionbudgets" in API group "policy" in the namespace "tenant-ns"`

**Review by shtripat — APPROVED** [review]
LGTM. Verified below scenarios using kustomize plugin
1. Create tenant (pdb creates)
2. Expand tenant (pdb cerates for new pool)
3. Decomm pool by editing YAML in console (pdb gets deleted for decomm'ed pool)
4. Delete tenant (all pdbs get deleted)

**Comment by jiuker** [comment]
@harshavardhana @pjuarezd  PATL

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**